code/modules/hydroponics/trays/tray.dm

```diff
@@ -472,7 +472,7 @@
 /obj/machinery/portable_atmospherics/hydroponics/CtrlClick(var/mob/user)
 	if(usr.incapacitated())
 		return
-	if(ishuman(usr) || istype(usr, /mob/living/silicon/robot))
+	if(ishuman(usr) && Adjacent(usr) || issilicon(usr))
```

Comment:
these checks are all covered by use_check (or use_check_and_message if you want a message to be sent on failure)